journal.py

```diff
@@ -133,68 +136,55 @@ def write_entry(request):
     title = request.params.get('title', None)
     text = request.params.get('text', None)
     created = datetime.datetime.utcnow()
-    request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
+    request.db.cursor().execute(add_entry, [title, text, created])
 
 
-@view_config(route_name='add', request_method='POST')
+@view_config(route_name='add', renderer='json', request_method="POST")
 def add_entry(request):
-    try:
-        write_entry(request)
-    except psycopg2.Error:
-        # this will catch any errors generated by the database
-        return HTTPInternalServerError
-    return HTTPFound(request.route_url('home'))
+    if request.authenticated_userid:
+        if request.method == 'POST':
+            try:
+                Entry.from_request(request)
+            except psycopg2.Error:
```

Comment:
This error will never be raised by SQLAlechemy.  You'll have to update the error handling here to catch SQLA errors.